### DISCUSSION TITLE
專家點出Python之父近期重大宣布，CPython執行速度4年內將翻5倍

### DISCUSSION THREAD

**Discussion by minniewei** [1101-GS3073-C/Python-Community-GS3073-C#23] [General]
根據新聞表示，python相較其他程式語言執行速度過慢，需要透過C語言編寫的直譯器CPython來加速Python的執行。但就算有Cpython的幫助，在進行龐大數據運算時，往往還需要搭配如Numpy、TensorFlow等以C語言為主編寫的函式庫，來提升執行效率。而現在有轉機了！Guido van Rossum在微軟開發者社群Reactor的一場採訪中表示，將與微軟4位Python核心開發者，一起研究提升CPython效能的方法，所以在未來pyhon會執行得更快，能進行更複雜的運算，甚至能超過C成為程式語言排行榜TIOBE Index的第一名！但我覺得最重要的是在python執行效率的增加下，說不定未來ai產業能有更大的突破！！期待看到如此美好的未來！！

https://www.ithome.com.tw/news/145695

**Comment by jim11261314** [comment]
這學期系上才剛教使用python分析資料，含糊地說會比EXCEL更快
沒想到原來C目前比python更快

**Comment by qeeeee39** [comment]
之前有看過人把fortran和python以包成module的方式導入結合（fortran跑很快但撰寫不直觀，python跑很慢但撰寫簡單），最後跑模擬數據的速度居然只比fortran慢一點點。

我還不了解cpython，但不同語言的相互優缺補足很有趣。

**Comment by Yunhans** [comment]
沒想到python會有這缺點

**Comment by LanningYang** [comment]
python是剛接觸到的，但確實聽說目前企業上大多使用C語言
